RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc

```diff
@@ -25,14 +34,35 @@ unsigned HFSimpleTimeCheck::determineAnodeStatus(
     const unsigned nRaw = anode.nRaw();
     if (nRaw)
     {
-        const QIE10DataFrame::Sample s(anode.getRaw(nRaw - 1));
-        if (!s.ok())
+        bool hardwareOK = true;
+        unsigned sampleToCheck = anode.soi();
+        if (sampleToCheck < nRaw)
+        {
+            const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
+            hardwareOK = s.ok();
+        }
+        else
+        {
+            // This should not normally happen, but we still
+            // need to do something reasonable here
+            for (sampleToCheck = 0; sampleToCheck < nRaw; ++sampleToCheck)
+            {
+                const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
+                if (!s.ok())
+                    hardwareOK = false;
+            }
+        }
+        if (!hardwareOK)
```

Comment:
@igv4321: OK. I was just making a suggestion. If this method is called at very high frequency, then performance considerations take priority.